File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Achievements/AchievementSelectedCategoryModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Achievements;

public class AchievementSelectedCategoryModel
{
    [JsonPropertyName("achievements")]
    public AchievementExtendModel[] Achievements { get; set; }

    [JsonPropertyName("subcategories")]
    public AchievementCategoryModel[] Subcategories { get; set; }

    [JsonPropertyName("is_guild_category")]
    public bool IsGuildCategory { get; set; }

    [JsonPropertyName("aggregates_by_faction")]
    public AchievementSelectedCategoryFactionModel AggregatesByFaction { get; set; }

    [JsonPropertyName("display_order")]
    public int DisplayOrder { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/achievements/AchievementSelectedCategoryModel.ts
import type { AchievementSelectedCategoryFactionModel } from './AchievementSelectedCategoryFactionModel';
import type { AchievementCategoryModel } from './AchievementCategoryModel';
import type { AchievementExtendModel } from './AchievementExtendModel';

export type AchievementSelectedCategoryModel = {
    achievements: AchievementExtendModel[];
    subcategories: AchievementCategoryModel[];
    isGuildCategory: boolean;
    aggregatesByFaction: AchievementSelectedCategoryFactionModel;
    displayOrder: number;
}